irohad/main/application.cpp

```diff
@@ -195,20 +195,21 @@ Irohad::RunResult Irohad::initSettings() {
   auto settingsQuery = storage->createSettingQuery();
   if (not settingsQuery) {
     return expected::makeError("Unable to create Settings");
-  } else {
-    auto settings = settingsQuery.get()->get();
-
-    if (auto e = boost::get<expected::Error<std::string>>(&settings)) {
-      return *e;
-    }
+  }
 
-    settings_ = boost::get<expected::Value<
-        std::shared_ptr<const shared_model::validation::Settings>>>(settings)
-                    .value;
+  auto settings = settingsQuery.get()->get();
 
-    log_->info("[Init] => settings");
-    return {};
+  if (auto e = boost::get<expected::Error<std::string>>(&settings)) {
+    return *e;
   }
+
+  settings_ = std::move(
+      boost::get<expected::Value<
+          std::unique_ptr<const shared_model::validation::Settings>>>(settings)
+          .value);
```

Comment:
```
  settings_ = std::move(expected::resultToOptionalValue(settings)).value();
```

Reply:
or even   return settingsQuery.get()->get() | [this](auto &&settings) -> RunResult {
```c++
  return settingsQuery.get()->get() | [this](auto &&settings) -> RunResult {
    this->settings_ = std::move(settings);

    log_->info("[Init] => settings");
    return {};
  };
```

Reply:
How to process errors in this case?

Reply:
Fixed